pkg/volume/host_path/host_path.go

```diff
@@ -98,13 +99,47 @@ func (plugin *hostPathPlugin) GetAccessModes() []v1.PersistentVolumeAccessMode {
 	}
 }
 
-func (plugin *hostPathPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, _ volume.VolumeOptions) (volume.Mounter, error) {
+func (plugin *hostPathPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, opts volume.VolumeOptions) (volume.Mounter, error) {
 	hostPathVolumeSource, readOnly, err := getVolumeSource(spec)
 	if err != nil {
 		return nil, err
 	}
+
+	path := hostPathVolumeSource.Path
+	// A containerized kubelet would have difficulty creating directories.
+	// The implementation will likely respect the containerized flag, allowing it to be "/rootfs/" aware and
+	// thus operate as desired.
+	if opts.Containerized {
+		path += "/rootfs" + path
+	}
+
+	_, err = os.Stat(path)
+	notExist := os.IsNotExist(err)
+	switch hostPathVolumeSource.Type {
+	case "":
+		if notExist {
+			err = os.Mkdir(path, os.ModePerm)
+			if err != nil {
+				return nil, err
+			}
+		}
+	case "exists":
+		if notExist {
+			return nil, fmt.Errorf("Nothing exists at %s",
+				hostPathVolumeSource.Path)
+		}
+	case "file", "device", "socket", "directory":
+		if notExist {
+			return nil, fmt.Errorf("%s does not exist at %s",
+				strings.Title(hostPathVolumeSource.Type),
+				hostPathVolumeSource.Path)
+		}
+	case "new-directory", "character-device", "block-device", "new-file", "optional":
+		return nil, fmt.Errorf("%s is invalid volume type", hostPathVolumeSource.Type)
```

Comment:
The [original proposal](https://github.com/kubernetes/kubernetes/pull/34058) listed `new-file` and `new-directory` in this space, but claimed they were not needed.

I claim `new-file` is needed, since it is sometimes unsafe to mount the parent directory.  Example `/var/run`, which is usually the parent of Docker container volumes, hence you get a recursive mount which tickles kernel bugs.